metals/src/main/scala/scala/meta/internal/builds/Digest.scala

```diff
@@ -0,0 +1,251 @@
+package scala.meta.internal.builds
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Path
+import java.security.MessageDigest
+import java.util.stream.Collectors
+
+import scala.meta.internal.builds.Digest.Status
+import scala.meta.internal.io.PathIO
+import scala.meta.internal.mtags.MD5
+import scala.meta.internal.mtags.MtagsEnrichments._
+import scala.meta.io.AbsolutePath
+import scala.meta.tokens.Token
+import scala.util.control.NonFatal
+import scala.collection.JavaConverters._
+
+case class Digest(
+    md5: String,
+    status: Status,
+    millis: Long
+)
+
+object Digest {
+
+  /**
+   * Bump up this version if parameters outside of the sbt sources themselves require
+   * re-running `bloopInstall`. For example a SemanticDB or Bloop version upgrade.
+   */
+  val version: String = "v4"
+  sealed abstract class Status(val value: Int)
```

Comment:
why not something like:
```scala
final class Status private(val value: Int){
  def ...
}

object Status{
  val Requested = new Status(0)
  val Started = new Status(1)
  ...
}
```

Reply:
This is basically the same and lacks ability to sensibly pattern match. I would leave it as is.